File: examples/false-positives/compose/idiomatic-state.kt
package com.example.billing

@Composable
fun BillingSummaryRow(
    uiState: BillingSummaryState,
    onAction: (BillingAction) -> Unit,
    modifier: Modifier = Modifier,
    leadingIcon: @Composable (() -> Unit)? = null,
    trailingContent: @Composable () -> Unit = {},
) {
    val latestAction by rememberUpdatedState(onAction)
    val listState = rememberLazyListState()
    /*
    var oldQuery by remember { mutableStateOf("") }
    var oldExpanded by remember { mutableStateOf(false) }
    */
    Row(modifier) {
        leadingIcon?.invoke()
        Text(uiState.customerName)
        Button(onClick = { latestAction(BillingAction.Open(uiState.invoiceId)) }) {
            Text("Open")
        }
        trailingContent()
    }
    LaunchedEffect(uiState.invoiceId) {
        listState.scrollToItem(0)
    }
}

File: examples/false-positives/swiftui/idiomatic-state.swift
import SwiftUI

struct BillingSummaryRow: View {
    let uiState: BillingSummaryState
    let onAction: (BillingAction) -> Void
    var leadingIcon: (() -> AnyView)?
    let trailingContent: () -> AnyView

    @State private var pulse = false

    var body: some View {
        HStack {
            if let leadingIcon {
                leadingIcon()
            }
            Text(uiState.customerName)
            Button("Open") {
                onAction(.open(uiState.invoiceId))
            }
            trailingContent()
        }
        .opacity(pulse ? 1 : 0.95)
    }
}

struct BillingSummaryState {
    let customerName: String
    let invoiceId: String
}

enum BillingAction {
    case open(String)
}
